rfds/502 - How to write an RFD/index.adoc

```diff
@@ -0,0 +1,80 @@
+= Writing a Request For Discussion
+Malcolm Matalka <malcolm@terrateam.io>
+2025-05-29
+:source-highlighter: highlight.js
+:toc:
+
+== Status
+
+In discussion
+
+== Introduction
+
+Communication is very important to Terrateam.  While many discussions are easy
+to happen over a meeting or in chat, some discussions require so much context
+and though to be effectively communicated that they require a long-form way to
+communicating.  Terrateam has chosen an RFD as the form this should take.
+
+RFDs are not unique to Terrateam.  The most prominent user of RFDs are Oxide.
+However the basic structure has been taken from the RFC system that has existed
+since the creation of the internet.
+
+== Why RFDs?
+
+We want to have RFDs for the following reasons:
+
+. Historical record - For decisions that we make that are of substance, it is
+useful to have a centralized store of them.  It also makes them searchable.
+. Larger context - Some ideas require more context to entirely appreciate than
+can be done in a meeting or in chat.  An RFD is the place where the author has
+the reader's undivided attention and is able to give them all of the context
+necessary to understand whatever is being discussed.
+
+
+== When to write an RFD
+
+There is no minimum requirement on when to write an RFD.  You should use your
+discretion on when writing one would be valuable.
+
+More importantly **do not** reject writing an RFD for any of these reasons:
+
+. The idea is not fully formed.  That is OK.  An RFD is a valid place to explore
+an idea and gather feedback.  The D stands for discussion.
+. The idea is too small.  That is OK.  Even small ideas can be important.
+
+In general, an RFD should almost definitely be written when:
+
+. There is a change in architecture to the product.  Getting feedback from
+multiple people improves the chances something will not be missed.
+. There is a change that will impact many parts of the system.  Getting feedback
+will help ensure that the change aligns with what other people are working on
+and their view of the system.
+
+== How to write an RFD
+
+. Every RFD must be associated with a ticket in the
+https://github.com/terrateamio/terrateam/issues[Terrateam] repository.
```

Comment:
I'd also add that we should make a practice to link existing accepted/rejected RFD's in the issues themselves, if that makes sense. This way we could derive a historical chain of decisions later. Similar to what happens with RFCs.

Reply:
If we add the issue number to the commit, does that do the linking you are thinking of, or is there another layer of linking you have in mind?

Reply:
I like the way it's being done on Oxide's side, with proper links to past RFDs (i.e. https://rfd.shared.oxide.computer/rfd/0419).

![image](https://github.com/user-attachments/assets/ced4de7a-709d-4905-81bd-fec6c2d2924c)

(Picture just in case they change this later)

I mentioned adding it on issues because it seems to be easier to build a graph out of it later, unless you have a different solution in mind.

Reply:
I added a little language around this.